packages/kbn-management/settings/utilities/setting/normalize_settings.ts

```diff
@@ -71,7 +71,7 @@ const deriveValue = (type: SettingType, value: unknown): Value => {
     case 'string':
       return value as string;
     case 'number':
-      return value ? Number(value) : undefined;
+      return value != null ? Number(value) : undefined;
```

Comment:
I think this change will change the output of this function for some values of `value`, e.g. `deriveValue('number', false)` will now return `0` instead of `undefined`.

Is the main goal of this casting to cast strings to numbers, probably? Maybe checking if `typeof` is number or string and doing the cast only then would be clearer/safer? 

This is outside the CODEOWNER path for our team so I won't block the review on this, just wanted to point it out since that `unknown` type can be a killer if you're not careful :)

Reply:
Do we unit test this? if not, it would be a good time to create something.

Reply:
That's a good call @jasonrhodes! Yes, we want to cast numeric strings to a number. I changed the way we do this in a safer way in [3bc0eee](https://github.com/elastic/kibana/pull/177716/commits/3bc0eee390d16695db8ed4d9f3ff485a9615c8b5). Also added some tests. Let me know if you have any other suggestions.